src/nvim/change.c

```diff
@@ -208,8 +208,7 @@ static void changed_lines_invalidate_win(win_T *wp, linenr_T lnum, colnr_T col,
           wp->w_lines[i].wl_foldend += xtra;
           wp->w_lines[i].wl_lastlnum += xtra;
         }
-      } else if (wp->w_lines[i].wl_foldend >= lnum
-                 || wp->w_lines[i].wl_lastlnum >= lnum) {
+      } else if (wp->w_lines[i].wl_lastlnum >= lnum) {
```

Comment:
Why is this change include in this PR?

Reply:
Dirty workspace when starting work on this🤷🏻. (Check was added in #32766 but can be simplified like this.)